### ISSUE TITLE
io: `bridge` uses function `inner`

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#3832]
Are we happy with the `FromStr::inner` function name (and `ToStd`) or should it be `as_inner` - it returns a reference.

**Comment by tcharding** [comment]
Note, it does pair with `inner_mut`

**Comment by tcharding** [comment]
Perhaps it should be `into_inner`, `get_ref`, and `get_mut`. Same as in #3823 (`std::io::Cursor`). Also same as `std::io::Take` https://doc.rust-lang.org/std/io/struct.Take.html

**Comment by apoelstra** [comment]
Hmm, the analogy with `Take` is pretty good. Yeah, I am tempted to move the API to that.

**Comment by tcharding** [comment]
Was reading the docs to `Box` today. The API there uses `into_inner`, `AsRef`, and `AsMut`.

ref: https://doc.rust-lang.org/std/boxed/struct.Box.html#method.into_inner

**Comment by apoelstra** [comment]
Hmm, `Box` is a smart pointer and isn't really analogous to a wrapper that meaningfully changes the API.

**Comment by tcharding** [comment]
Fair. I cannot work out why `std::io` use `fn get_ref` instead of `convert::AsRef` but grepping the stdlib gives

```bash
ack 'fn get_ref'

core/src/pin.rs
1564:    pub const fn get_ref(self) -> &'a T {

std/src/sync/poison.rs
247:    pub fn get_ref(&self) -> &T {

std/src/io/cursor.rs
135:    pub const fn get_ref(&self) -> &T {

std/src/io/mod.rs
2617:    pub fn get_ref(&self) -> (&T, &U) {
2876:    pub fn get_ref(&self) -> &T {

std/src/io/error.rs
736:    pub fn get_ref(&self) -> Option<&(dyn error::Error + Send + Sync + 'static)> {

std/src/io/buffered/bufwriter.rs
283:    pub fn get_ref(&self) -> &W {

std/src/io/buffered/linewriter.rs
185:    pub fn get_ref(&self) -> &W {

std/src/io/buffered/bufreader.rs
171:    pub fn get_ref(&self) -> &R {

```

Since these are all but one found in the `io` module perhaps we should just do the same (since we are discussing our `io` module(?

**Comment by apoelstra** [comment]
Yeah, I think we should copy `std::io` to the extent possible. Ideally users can just s/bitcoin::io/std::io/ and vice-versa to switch between them.
